### ISSUE TITLE
pyexiv2: stable 0.3.2 crashes with SIGSEGV at import

### ISSUE THREAD

**Issue by primiano** [Homebrew/homebrew-core#7735]
- [X] Ran `brew update` and retried your prior step?

- [X] Ran `brew doctor`, fixed as many issues as possible and retried your prior step

- [X] Confirmed this is problem with specific formulae and not Homebrew/brew? 

### Bug reports:
importing pyexiv2 fails with a segfault. Using latest python and pyexiv from homebrew.

```
$ python
Python 2.7.12 (default, Oct 11 2016, 05:24:00)
>>> import pyexiv2
Segmentation fault: 11
```

```
$ brew info python
python: stable 2.7.12 (bottled), HEAD
Interpreted, interactive, object-oriented programming language
https://www.python.org
/usr/local/Cellar/python/2.7.9 (4,999 files, 69.7M)
  Poured from bottle on 2014-12-19 at 06:06:37
/usr/local/Cellar/python/2.7.12_2 (3,546 files, 48M) *
  Poured from bottle on 2016-12-09 at 22:33:39
```

```
$ brew info pyexiv2
pyexiv2: stable 0.3.2 (bottled)
Python binding to exiv2 for manipulation of image metadata
http://tilloy.net/dev/pyexiv2/
/usr/local/Cellar/pyexiv2/0.3.2_1 (14 files, 574K) *
  Poured from bottle on 2016-12-10 at 00:12:00
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/pyexiv2.rb
==> Dependencies
Build: scons ✔
Required: exiv2 ✔, boost ✔, boost-python ✔
```

Crash log:
```
Process:               Python [10171]
Path:                  /usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python
Identifier:            Python
Version:               2.7.12 (2.7.12)
Code Type:             X86-64 (Native)
Parent Process:        bash [10050]
Responsible:           iTerm2 [68150]
User ID:               501

Date/Time:             2016-12-10 00:15:21.880 +0000
OS Version:            Mac OS X 10.11.6 (15G1108)
Report Version:        11
Anonymous UUID:        2D3DB181-20C0-6D63-5D9B-EA62F8B67B43

Sleep/Wake UUID:       F272F4D0-1BFA-48C3-B8A7-76063E0F6C6E

Time Awake Since Boot: 170000 seconds
Time Since Wake:       16000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000000

VM Regions Near 0:
--> 
    __TEXT                 0000000109bfd000-0000000109bff000 [    8K] r-x/rwx SM=COW  /usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python

Application Specific Information:
/usr/local/lib/python2.7/site-packages/libexiv2python.so

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   org.python.python             	0x000000010a4ffc81 PyInt_FromLong + 33
1   libexiv2python.so             	0x000000010a1c1652 _object* boost::python::api::object_initializer_impl<false, false>::get<int>(int const&, mpl_::bool_<false>) + 14
2   libexiv2python.so             	0x000000010a1c1478 boost::python::tuple boost::python::make_tuple<int, int, int>(int const&, int const&, int const&) + 60
3   libexiv2python.so             	0x000000010a1cc084 _GLOBAL__sub_I_exiv2wrapper_python.cpp + 99
4   dyld                          	0x00007fff6b0c210b ImageLoaderMachO::doModInitFunctions(ImageLoader::LinkContext const&) + 265
5   dyld                          	0x00007fff6b0c2284 ImageLoaderMachO::doInitialization(ImageLoader::LinkContext const&) + 40
6   dyld                          	0x00007fff6b0be8bd ImageLoader::recursiveInitialization(ImageLoader::LinkContext const&, unsigned int, ImageLoader::InitializerTimingList&, ImageLoader::UninitedUpwards&) + 305
7   dyld                          	0x00007fff6b0be743 ImageLoader::processInitializers(ImageLoader::LinkContext const&, unsigned int, ImageLoader::InitializerTimingList&, ImageLoader::UninitedUpwards&) + 127
8   dyld                          	0x00007fff6b0be9b3 ImageLoader::runInitializers(ImageLoader::LinkContext const&, ImageLoader::InitializerTimingList&) + 75
9   dyld                          	0x00007fff6b0b3eb0 dyld::runInitializers(ImageLoader*) + 89
10  dyld                          	0x00007fff6b0bb308 dlopen + 555
11  libdyld.dylib                 	0x00007fff82b2c79c dlopen + 59
12  org.python.python             	0x0000000109cbb182 _PyImport_GetDynLoadFunc + 309
13  org.python.python             	0x0000000109ca6230 _PyImport_LoadDynamicModule + 96
14  org.python.python             	0x0000000109ca4f58 import_submodule + 267
15  org.python.python             	0x0000000109ca4b6d load_next + 345
16  org.python.python             	0x0000000109ca3d03 PyImport_ImportModuleLevel + 1129
17  org.python.python             	0x0000000109c85a0e builtin___import__ + 135
18  org.python.python             	0x0000000109c119b4 PyObject_Call + 99
19  org.python.python             	0x0000000109c94677 PyEval_CallObjectWithKeywords + 165
20  org.python.python             	0x0000000109c8f7b9 PyEval_EvalFrameEx + 20917
21  org.python.python             	0x0000000109c8a40e PyEval_EvalCodeEx + 1617
22  org.python.python             	0x0000000109c89db7 PyEval_EvalCode + 48
23  org.python.python             	0x0000000109ca27e8 PyImport_ExecCodeModuleEx + 241
24  org.python.python             	0x0000000109ca54f3 load_source_module + 1095
25  org.python.python             	0x0000000109ca577b load_package + 303
26  org.python.python             	0x0000000109ca4f58 import_submodule + 267
27  org.python.python             	0x0000000109ca4b2c load_next + 280
28  org.python.python             	0x0000000109ca3d03 PyImport_ImportModuleLevel + 1129
29  org.python.python             	0x0000000109c85a0e builtin___import__ + 135
30  org.python.python             	0x0000000109c119b4 PyObject_Call + 99
31  org.python.python             	0x0000000109c94677 PyEval_CallObjectWithKeywords + 165
32  org.python.python             	0x0000000109c8f7b9 PyEval_EvalFrameEx + 20917
33  org.python.python             	0x0000000109c8a40e PyEval_EvalCodeEx + 1617
34  org.python.python             	0x0000000109c89db7 PyEval_EvalCode + 48
35  org.python.python             	0x0000000109cad83c run_mod + 53
36  org.python.python             	0x0000000109cad66a PyRun_InteractiveOneFlags + 377
37  org.python.python             	0x0000000109cad15e PyRun_InteractiveLoopFlags + 192
38  org.python.python             	0x0000000109cad008 PyRun_AnyFileExFlags + 60
39  org.python.python             	0x0000000109cbed9e Py_Main + 3094
40  libdyld.dylib                 	0x00007fff82b2d5ad start + 1

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x000000010a603060  rbx: 0x0000000000000000  rcx: 0x0000000000000000  rdx: 0x0000000000000000
  rdi: 0x0000000000000000  rsi: 0x0000000000000000  rbp: 0x00007fff56000cf0  rsp: 0x00007fff56000ce0
   r8: 0x00000000ffffffff   r9: 0x0000000000000000  r10: 0x0000000000001002  r11: 0x0000000000000206
  r12: 0x00007fff56000d68  r13: 0x00007fff56000d6c  r14: 0x000000010a1da728  r15: 0x00007fff56000d64
  rip: 0x000000010a4ffc81  rfl: 0x0000000000010287  cr2: 0x0000000000000000
  
Logical CPU:     2
Error Code:      0x00000006
Trap Number:     14


Binary Images:
       0x109bfd000 -        0x109bfefff +org.python.python (2.7.12 - 2.7.12) <90F2CB2F-F374-3FAB-9EEA-BA416C2D34FC> /usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python
       0x109c07000 -        0x109cfcff7 +org.python.python (2.7.12, [c] 2001-2016 Python Software Foundation. - 2.7.12) <F5D1C076-BE07-3D14-A39C-24A7C7D42F79> /usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/Python
       0x10a12c000 -        0x10a12efff +_locale.so (0) <DEAD0177-587C-32C8-84F2-E68A84F2F832> /usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-dynload/_locale.so
       0x10a131000 -        0x10a133fff +readline.so (0) <84E2015C-6051-3463-A8B8-54C530B0F549> /usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-dynload/readline.so
       0x10a139000 -        0x10a15cff3 +libreadline.7.dylib (0) <88BBCBEE-F3A9-3A5E-BE5E-16510B99EA23> /usr/local/opt/readline/lib/libreadline.7.dylib
       0x10a1b4000 -        0x10a1d6ffb +libexiv2python.so (0) <002644AA-E3AF-3826-BC56-EB2EB82C410B> /usr/local/lib/python2.7/site-packages/libexiv2python.so
       0x10a225000 -        0x10a253ff7 +libboost_python-mt.dylib (0) <D124D741-38CE-3D2C-BD05-571E8913BA64> /usr/local/opt/boost-python/lib/libboost_python-mt.dylib
       0x10a293000 -        0x10a40cff7 +libexiv2.14.dylib (0) <C65F8D41-BDFD-37D5-850F-DE47D5EE9EF7> /usr/local/opt/exiv2/lib/libexiv2.14.dylib
       0x10a4d7000 -        0x10a5c8ff7  org.python.python (2.7.10 - 2.7.10) <83AFAAA7-BDFA-354D-8A7A-8F40A30ACB91> /System/Library/Frameworks/Python.framework/Versions/2.7/Python
       0x10a62f000 -        0x10a648fff  libexpat.1.dylib (12.60.1) <6BF318AC-F90F-3F16-99C2-6ADDE404E280> /usr/lib/libexpat.1.dylib
    0x7fff6b0af000 -     0x7fff6b0e6a47  dyld (360.22) <DC81CC9D-651A-3A45-8809-928282052BD3> /usr/lib/dyld
    0x7fff803d8000 -     0x7fff804caff7  libiconv.2.dylib (44) <F05A0A5A-92A9-3668-8F20-F27CBDA26BE9> /usr/lib/libiconv.2.dylib
    0x7fff81c0c000 -     0x7fff81c72ff7  libsystem_network.dylib (583.50.1) <B52DAB73-92DC-3DA7-B9F4-B899D66445C1> /usr/lib/system/libsystem_network.dylib
    0x7fff82048000 -     0x7fff8204fff7  libcompiler_rt.dylib (62) <A13ECF69-F59F-38AE-8609-7B731450FBCD> /usr/lib/system/libcompiler_rt.dylib
    0x7fff82b2a000 -     0x7fff82b2dffb  libdyld.dylib (360.22) <25D28225-99AE-3A04-B98C-ACFF47D6D9B0> /usr/lib/system/libdyld.dylib
    0x7fff8383d000 -     0x7fff83859ff7  libsystem_malloc.dylib (67.40.1) <5748E8B2-F81C-34C6-8B13-456213127678> /usr/lib/system/libsystem_malloc.dylib
    0x7fff860dc000 -     0x7fff8610dff7  libncurses.5.4.dylib (46) <D4C10699-3E8F-3F42-A695-7490EA4B6B32> /usr/lib/libncurses.5.4.dylib
    0x7fff87333000 -     0x7fff8733cff3  libsystem_notify.dylib (150.40.1) <D48BDE34-0F7E-34CA-A0FF-C578E39987CC> /usr/lib/system/libsystem_notify.dylib
    0x7fff873a0000 -     0x7fff873a0ff7  libunc.dylib (29) <DDB1E947-C775-33B8-B461-63E5EB698F0E> /usr/lib/system/libunc.dylib
    0x7fff8858d000 -     0x7fff88595fff  libsystem_networkextension.dylib (385.40.36) <66095DC7-6539-38F2-95EE-458F15F6D014> /usr/lib/system/libsystem_networkextension.dylib
    0x7fff8872c000 -     0x7fff88755ff7  libxpc.dylib (765.50.8) <54D1328E-054E-3DAA-89E2-375722F9D18F> /usr/lib/system/libxpc.dylib
    0x7fff896e1000 -     0x7fff89734ff7  libc++.1.dylib (120.1) <8FC3D139-8055-3498-9AC5-6467CB7F4D14> /usr/lib/libc++.1.dylib
    0x7fff8ab92000 -     0x7fff8ac09feb  libcorecrypto.dylib (335.50.1) <B5C05FD7-A540-345A-87BF-8E41848A3C17> /usr/lib/system/libcorecrypto.dylib
    0x7fff8bd60000 -     0x7fff8c0c2f3f  libobjc.A.dylib (680) <7489D2D6-1EFD-3414-B18D-2AECCCC90286> /usr/lib/libobjc.A.dylib
    0x7fff8c64f000 -     0x7fff8c666ff7  libsystem_asl.dylib (323.50.1) <41F8E11F-1BD0-3F1D-BA3A-AA1577ED98A9> /usr/lib/system/libsystem_asl.dylib
    0x7fff8ce1f000 -     0x7fff8ce27ffb  libsystem_dnssd.dylib (625.60.4) <80189998-32B0-316C-B5C5-53857486713D> /usr/lib/system/libsystem_dnssd.dylib
    0x7fff8d5ef000 -     0x7fff8d619ff7  libc++abi.dylib (307.2) <922EFB36-0E9E-315B-8270-E81AC43472C0> /usr/lib/libc++abi.dylib
    0x7fff8dc38000 -     0x7fff8de45fff  libicucore.A.dylib (551.51.4) <3899B146-3840-3D4A-8C4A-FE391D5D25C7> /usr/lib/libicucore.A.dylib
    0x7fff8de46000 -     0x7fff8de49fff  libsystem_sandbox.dylib (460.60.2) <2A68B39C-B786-3A05-87A2-56E688469FB8> /usr/lib/system/libsystem_sandbox.dylib
    0x7fff8ec56000 -     0x7fff8ec5fff7  libsystem_pthread.dylib (138.10.4) <3DD1EF4C-1D1B-3ABF-8CC6-B3B1CEEE9559> /usr/lib/system/libsystem_pthread.dylib
    0x7fff8ef49000 -     0x7fff8ef4afff  libDiagnosticMessagesClient.dylib (100) <4243B6B4-21E9-355B-9C5A-95A216233B96> /usr/lib/libDiagnosticMessagesClient.dylib
    0x7fff8eff0000 -     0x7fff8eff5ff7  libmacho.dylib (875.1) <318264FA-58F1-39D8-8285-1F6254EE410E> /usr/lib/system/libmacho.dylib
    0x7fff8f9aa000 -     0x7fff8f9bbff7  libz.1.dylib (61.20.1) <B3EBB42F-48E3-3287-9F0D-308E04D407AC> /usr/lib/libz.1.dylib
    0x7fff8fa9a000 -     0x7fff8fa9bfff  libsystem_blocks.dylib (65) <1244D9D5-F6AA-35BB-B307-86851C24B8E5> /usr/lib/system/libsystem_blocks.dylib
    0x7fff904c5000 -     0x7fff904e3ff7  libsystem_kernel.dylib (3248.60.11) <A3F069EA-6A41-35CA-BDF8-187E0209B3E2> /usr/lib/system/libsystem_kernel.dylib
    0x7fff90595000 -     0x7fff9059dfff  libcopyfile.dylib (127) <A48637BC-F3F2-34F2-BB68-4C65FD012832> /usr/lib/system/libcopyfile.dylib
    0x7fff90d90000 -     0x7fff90db9fff  libsystem_info.dylib (477.50.4) <FAA9226D-64DE-3769-A6D8-6CABA4B7FF4D> /usr/lib/system/libsystem_info.dylib
    0x7fff91275000 -     0x7fff912bbff7  libauto.dylib (186) <999E610F-41FC-32A3-ADCA-5EC049B65DFB> /usr/lib/libauto.dylib
    0x7fff91846000 -     0x7fff918d3fef  libsystem_c.dylib (1082.60.1) <28733D22-553E-3CBC-8D2C-EDCEB46E46AF> /usr/lib/system/libsystem_c.dylib
    0x7fff918d4000 -     0x7fff918d4ff7  liblaunch.dylib (765.50.8) <834ED605-5114-3641-AA4D-ECF31B801C50> /usr/lib/system/liblaunch.dylib
    0x7fff92404000 -     0x7fff92431fff  libdispatch.dylib (501.40.12) <C7499857-61A5-3D7D-A5EA-65DCC8C3DF92> /usr/lib/system/libdispatch.dylib
    0x7fff9280f000 -     0x7fff92810ffb  libremovefile.dylib (41) <552EF39E-14D7-363E-9059-4565AC2F894E> /usr/lib/system/libremovefile.dylib
    0x7fff92812000 -     0x7fff92813ffb  libSystem.B.dylib (1226.10.1) <012548CD-614D-3AF0-B3B1-676F427D2CD6> /usr/lib/libSystem.B.dylib
    0x7fff928ad000 -     0x7fff928c4ff7  libsystem_coretls.dylib (83.40.5) <C90DAE38-4082-381C-A185-2A6A8B677628> /usr/lib/system/libsystem_coretls.dylib
    0x7fff929f8000 -     0x7fff92e6eff7  com.apple.CoreFoundation (6.9 - 1259) <51D1F34F-9FEC-3EF2-8382-2D13D75F845A> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
    0x7fff92e6f000 -     0x7fff92e74ff3  libunwind.dylib (35.3) <F6EB48E5-4D12-359A-AB54-C937FBBE9043> /usr/lib/system/libunwind.dylib
    0x7fff93ac1000 -     0x7fff93ac3fff  libsystem_coreservices.dylib (19.2) <1B3F5AFC-FFCD-3ECB-8B9A-5538366FB20D> /usr/lib/system/libsystem_coreservices.dylib
    0x7fff94426000 -     0x7fff94428ff7  libquarantine.dylib (80) <0F4169F0-0C84-3A25-B3AE-E47B3586D908> /usr/lib/system/libquarantine.dylib
    0x7fff94971000 -     0x7fff949a0ffb  libsystem_m.dylib (3105) <08E1A4B2-6448-3DFE-A58C-ACC7335BE7E4> /usr/lib/system/libsystem_m.dylib
    0x7fff94a54000 -     0x7fff94a5cfef  libsystem_platform.dylib (74.40.2) <29A905EF-6777-3C33-82B0-6C3A88C4BA15> /usr/lib/system/libsystem_platform.dylib
    0x7fff95c36000 -     0x7fff95c3afff  libcache.dylib (75) <9548AAE9-2AB7-3525-9ECE-A2A7C4688447> /usr/lib/system/libcache.dylib
    0x7fff95c3b000 -     0x7fff95c4cff7  libsystem_trace.dylib (201.10.3) <E9311C03-9E61-3B13-AF3F-A64956FFF269> /usr/lib/system/libsystem_trace.dylib
    0x7fff95f2e000 -     0x7fff95f2eff7  libkeymgr.dylib (28) <8371CE54-5FDD-3CE9-B3DF-E98C761B6FE0> /usr/lib/system/libkeymgr.dylib
    0x7fff96924000 -     0x7fff96926ff7  libsystem_configuration.dylib (802.40.13) <3DEB7DF9-6804-37E1-BC83-0166882FF0FF> /usr/lib/system/libsystem_configuration.dylib
    0x7fff96927000 -     0x7fff96928fff  libsystem_secinit.dylib (20) <32B1A8C6-DC84-3F4F-B8CE-9A52B47C3E6B> /usr/lib/system/libsystem_secinit.dylib
    0x7fff96974000 -     0x7fff9697fff7  libcommonCrypto.dylib (60075.50.1) <93732261-34B4-3914-B7A2-90A81A182DBA> /usr/lib/system/libcommonCrypto.dylib

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 123753
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=112.0M resident=0K(0%) swapped_out_or_unallocated=112.0M(100%)
Writable regions: Total=48.6M written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=48.6M(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Activity Tracing                  2048K        2 
Kernel Alloc Once                    4K        2 
MALLOC                            36.4M       10 
MALLOC guard page                   16K        4 
STACK GUARD                       56.0M        2 
Stack                             8192K        2 
VM_ALLOCATE                       1796K        6 
__DATA                            3908K       61 
__LINKEDIT                        92.8M       13 
__TEXT                            19.2M       57 
__UNICODE                          552K        2 
shared memory                        8K        3 
===========                     =======  ======= 
TOTAL                            220.5M      152 

Model: MacBookPro11,1, BootROM MBP111.0138.B17, 2 processors, Intel Core i5, 2,6 GHz, 8 GB, SMC 2.16f68
```

**Comment by MikeMcQuaid** [comment]
Any thoughts here @tdsmith?

**Comment by tdsmith** [comment]
Yeah, pyexiv2 is broken:

```
==> Installing pyexiv2
==> Downloading https://homebrew.bintray.com/bottles/pyexiv2-0.3.2_1.sierra.bottle.tar.gz
######################################################################## 100.0%
==> Pouring pyexiv2-0.3.2_1.sierra.bottle.tar.gz
Warning: python modules have explicit framework links
These python extension modules were linked directly to a Python
framework binary. They should be linked with -undefined dynamic_lookup
instead of -lpython or -framework Python.
  /usr/local/Cellar/pyexiv2/0.3.2_1/lib/python2.7/site-packages/libexiv2python.so
```

**Comment by MikeMcQuaid** [comment]
@tdsmith Disable the bottle temporarily?

**Comment by tdsmith** [comment]
Fixed by #7772.
